src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+    }
+
+    function isDoneWithLimitedReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.LIMITED_REPORTING;
+    }
+
+    function isDoneWithAllReporters() public constant returns (bool) {
+        return getReportingState() > ReportingState.ALL_REPORTING;
+    }
+
+    function isFinalized() public constant returns (bool) {
+        return finalPayoutDistributionHash != bytes32(0);
+    }
+
+    function getFinalizationTime() public constant returns (uint256) {
+        return finalizationTime;
+    }
+
+    function isInAutomatedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_REPORTING;
+    }
+
+    function isInAutomatedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.AUTOMATED_DISPUTE;
+    }
+
+    function isInLimitedReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_REPORTING;
+    }
+
+    function isInLimitedDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.LIMITED_DISPUTE;
+    }
+
+    function isInAllReportingPhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_REPORTING;
+    }
+
+    function isInAllDisputePhase() public constant returns (bool) {
+        return getReportingState() == ReportingState.ALL_DISPUTE;
+    }
+
+    function isContainerForReportingToken(ReportingToken _shadyToken) public constant returns (bool) {
+        if (address(_shadyToken) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_shadyToken.getTypeName() != "ReportingToken") {
+            return false;
+        }
+        bytes32 _shadyId = _shadyToken.getPayoutDistributionHash();
+        ReportingToken _reportingToken = reportingTokens[_shadyId];
+        if (address(_reportingToken) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_reportingToken != _shadyToken) {
+            return false;
+        }
+        return true;
+    }
+
+    function isContainerForShareToken(IShareToken _shadyShareToken) public constant returns (bool) {
+        if (_shadyShareToken.getTypeName() != "ShareToken") {
+            return false;
+        }
+        return(getShareToken(_shadyShareToken.getOutcome()) == _shadyShareToken);
+    }
+
+    function isContainerForDisputeBondToken(DisputeBondToken _shadyBondToken) public constant returns (bool) {
+        if (_shadyBondToken.getTypeName() != "DisputeBondToken") {
+            return false;
+        }
+        if (automatedReporterDisputeBondToken == _shadyBondToken) {
+            return true;
+        } else if (limitedReportersDisputeBondToken == _shadyBondToken) {
+            return true;
+        } else if (allReportersDisputeBondToken == _shadyBondToken) {
+            return true;
+        }
```

Comment:
Style: Recommend just `if(...) { return ... }` rather than `else if`.  They are functionally the same when a return is involved, but I find clear guard clauses to be slightly more readable.